apps/mobile/src/components/Login/AuthProvider/Farcaster/FarcasterAuthProvider.tsx

```diff
@@ -182,19 +192,44 @@ export function useLoginWithFarcaster() {
     nonce,
   });
 
-  const initiateConnection = useCallback(async () => {
-    if (!isConnected) {
-      const { nonce } = await createNonce();
-      setNonce(nonce);
+  const initiateConnection = useCallback(
+    async (shouldAttemptReconnect = false) => {
+      setIsFarcasterLoading(true);
 
-      if (isConnectError) {
+      if (shouldAttemptReconnect && !isSuccess && url) {
         reconnect();
+        const { nonce } = await createNonce();
+        setNonce(nonce);
+        signIn();
+        Linking.openURL(url);
```

Comment:
if we're attempting to reconnect i don't think we need to create a new nonce! so we can probably remove L201-202